### PR TITLE
GEODE-4171: Putting in changes to fix and clarify the BUILDING.md

### PR DESCRIPTION
**Pull request by igodwin**
Fixes to the BUILDING.md.

- Removed src from cmake command, and added GEODE_ROOT build option
- Added Apache Geode binaries to Required Tools section

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
     $ cd <clone>
     $ mkdir build
     $ cd build
-    $ cmake ../src
+    $ cmake .. -DGEODE_ROOT=<path to Apache Geode binaries>
```
Comment (pivotal-jbarrett) [inline BUILDING.md]: This is optional and only necessary if Geode is not installed.

**Comment by igodwin** [comment]
Made suggested changes.

File: BUILDING.md [hunk 2]
```diff
@@ -15,9 +16,22 @@
...
-    $ cmake ../src
+    $ cmake ..
     $ cmake --build . -- /m
 
+## Optional Build Parameters
+
+-DGEODE_ROOT=<path to Apache Geode binaries>
+ This will allow you to specify the path to use for the Apache Geode libraries and binaries
+
+On Windows
```
Comment (pivotal-jbarrett) [inline BUILDING.md]: This is not relevant to CMake's configure phase like -D options above are. You should distinguish between the two.  Be careful you don't re-invent CMake docs.

File: BUILDING.md [hunk 3]
```diff
@@ -15,9 +16,22 @@
...
+
+ /m Will allow for the parallelization the build and link process
+
+On Linux
+
+ -j <# of parallel processes>  Will allow for the parallelization of the build and link processes
```
Comment (pivotal-jbarrett) [inline BUILDING.md]: Again is a build param.

File: BUILDING.md [hunk 4]
```diff
@@ -4,34 +4,43 @@
...
     $ mkdir build
     $ cd build
-    $ cmake ../src
+    $ cmake ..
     $ cmake --build . -- /m
 
-### [Windows System Requirements](#windows)
+## Optional Build Parameters
+
+On Windows
```
Comment (pivotal-jbarrett) [inline BUILDING.md]: Use MD formatting here.
```md
### Windows
```

File: BUILDING.md [hunk 5]
```diff
@@ -4,34 +4,43 @@
...
+
+ /m Will allow for the parallelization of the build and link process
+
+On Linux
```
Comment (pivotal-jbarrett) [inline BUILDING.md]:
```md
### Unix / Makefile
```

**Pull request merged**